include/atomics/subnetCadmium.hpp

```diff
@@ -28,86 +28,96 @@ using namespace cadmium;
+                                 typename make_message_bags<input_ports>::type mbs) { 
+            state.index ++;
+            if (get_messages<typename defs::in>(mbs).size() > 1) {
+                assert(false && "One message at a time");                
             }
+            for (const auto &x : get_messages<typename defs::in>(mbs)) {
+                state.packet = static_cast<int>(x.value);
+                state.transmiting = true; 
+            }               
+        }
+
+        // confluence transition
+        void confluence_transition(TIME e, 
+                                   typename make_message_bags<input_ports>::type mbs) {
+            internal_transition();
+            external_transition(TIME(), std::move(mbs));
+        }
 
-            // output function
-            typename make_message_bags<output_ports>::type output() const {
-              typename make_message_bags<output_ports>::type bags;
-              Message_t out;
-              if ((double)rand() / (double) RAND_MAX  < 0.95){
+        // output function
+        typename make_message_bags<output_ports>::type output() const {
+            typename make_message_bags<output_ports>::type bags;
+            message_t out;
+            if ((double)rand()/(double)RAND_MAX  < 0.95) {
                 out.value = state.packet;
                 get_messages<typename defs::out>(bags).push_back(out);
-              }
-              return bags;
             }
+            return bags;
+        }
 
-            // time_advance function
-            TIME time_advance() const {
-              std::default_random_engine generator;
-              std::normal_distribution<double> distribution(3.0, 1.0); 
-              TIME next_internal;
-              if (state.transmiting) {
-                std::initializer_list<int> time = {0, 0, static_cast < int > (round(distribution(generator)))};
+        // time_advance function
+        TIME time_advance() const {
+            std::default_random_engine generator;
+            std::normal_distribution<double> distribution(3.0, 1.0); 
+            TIME next_internal;
+            if (state.transmiting) {
+                std::initializer_list<int> time 
+                    = {0, 0, static_cast<int>(round(distribution(generator)))};
                 // time is hour min and second
                 next_internal = TIME(time);
-              }else {
-                next_internal = std::numeric_limits<TIME>::infinity();
-              }    
-              return next_internal;
             }
+            else {
+                next_internal = std::numeric_limits<TIME>::infinity();
+            }    
+            return next_internal;
+        }
 
-            friend std::ostringstream& operator<<(std::ostringstream& os, const typename Subnet<TIME>::state_type& i) {
-                os << "index: " << i.index << " & transmiting: " << i.transmiting; 
+        friend std::ostringstream& operator<<(std::ostringstream& os, 

```

Comment:
Again, make sure that variable names are self-explanatory.

Reply:
Many thanks for letting us know about the variable names. We have fixed it now. May I request you to please review them ?